### PR TITLE
Change apache http instrumentation into command line

### PR DESCRIPTION
**Pull request by changexd**
**Description:** <Describe what has changed.>

change apache http instrumentation into command line

**Link to tracking Issue(s):** <Issue number if applicable>
- Resolves: #2670 

**Testing:** <Describe what testing was performed and which tests were added.>
I've removed `setFeatureGates`  and use config to enable instrumentation for apache http tests

**documentation**
this must be merge after #2635, unlike multi-instrumentation, moving featuregate into config also changes `InstrumentationUpgrade` as I wasn't sure if I handled this properly, feel free letting me know if anything needs to be changed, I'll then work on others one by one in the same way after this is accepted.

### CODE REVIEW

**Review by swiatekm-sumo — COMMENTED** [review]
Shouldn't we also remove the featuregate? After this change, it doesn't do anything.

File: .chloggen/apache-http-instrumentation-command-flag.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. operator, target allocator, github action)
+component: operator
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: change apache http instrumentation feature gate into command line flag --enable-apache-http-instrumentation
```
Comment (TylerHelmuth) [inline .chloggen/apache-http-instrumentation-command-flag.yaml:8]: Let's specifically call out the name of the feature gate that is being removed

File: pkg/constants/env.go [hunk 2]
```diff
@@ -34,4 +34,6 @@ const (
 	EnvPodName  = "OTEL_RESOURCE_ATTRIBUTES_POD_NAME"
 	EnvPodUID   = "OTEL_RESOURCE_ATTRIBUTES_POD_UID"
 	EnvNodeName = "OTEL_RESOURCE_ATTRIBUTES_NODE_NAME"
+
+	ApacheHttpd = "apache-httpd"
```
Comment (TylerHelmuth) [inline pkg/constants/env.go:38]: Is this being used?

File: pkg/instrumentation/upgrade/upgrade.go [hunk 3]
```diff
@@ -52,6 +58,28 @@ type InstrumentationUpgrade struct {
 	DefaultAutoInstApacheHttpd string
 	DefaultAutoInstNginx       string
 	DefaultAutoInstGo          string
+	defaultAnnotationToConfig  map[string]autoInstConfig
+}
+
+func NewInstrumentationUpgrade(client client.Client, recorder record.EventRecorder, cfg config.Config) *InstrumentationUpgrade {
```
Comment (TylerHelmuth) [inline pkg/instrumentation/upgrade/upgrade.go:64]: This function is only used in tests (that I can see).  Does it need included here?

Comment (changexd) [reply pkg/instrumentation/upgrade/upgrade.go:64]: Sorry I forgot to put this in `main.go` after the change, it was supposed to move initialization of `InstrumentationUpgrade` to this function, will do it now

File: pkg/instrumentation/upgrade/upgrade.go [hunk 4]
```diff
@@ -52,6 +58,28 @@ type InstrumentationUpgrade struct {
...
+	defaultAnnotationToConfig := map[string]autoInstConfig{
+		constants.AnnotationDefaultAutoInstrumentationApacheHttpd: autoInstConfig{constants.ApacheHttpd, cfg.EnableApacheHTTPAutoInstrumentation()},
```
Comment (changexd) [inline pkg/instrumentation/upgrade/upgrade.go:66]: @TylerHelmuth `constants.ApacheHttpd` is used here for mapping corresponding annotation, it previously use `featuregate.EnableApacheHttpAutoInstrumentationSupport.ID()` for logging when a featuregate is disabled, I then follow the same way how `annotations` are mapped to `featuregate`.

Comment (TylerHelmuth) [reply pkg/instrumentation/upgrade/upgrade.go:65]: I think we should be logging the full command line flag (`enable-apache-http-instrumentation`) like the feature gate did.  `apache-httpd` is not the name of the flag

File: .chloggen/apache-http-instrumentation-command-flag.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. operator, target allocator, github action)
+component: operator
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note:  remove featuregate "EnableApacheHTTPAutoInstrumentationSupport" in replacement of command line flag --enable-apache-httpd-instrumentation
```
Comment (TylerHelmuth) [inline .chloggen/apache-http-instrumentation-command-flag.yaml:8]:
```suggestion
note:  remove featuregate `EnableApacheHTTPAutoInstrumentationSupport`. Use command line flag `--enable-apache-httpd-instrumentation` instead
```

**Review by TylerHelmuth — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Comment by jaronoff97** [comment]
@changexd thank you for your contribution! I made issues for each language and updated the PR to use the new issues. This will merge when tests succeed.

**Pull request merged**